### PR TITLE
adding policy yaml structure in the document

### PR DESCRIPTION
**Pull request by rutu-k**

### CODE REVIEW

File: web/scripts/display_yaml.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Input parametes for creating the policy yaml
+
+let source_label = {app: "test"};
+let target_label = {app: "final"};
```
Comment (viveksinghggits) [inline web/scripts/display_yaml.js]: there are chances that we are going to select set of labels for target and what id the source already has set of labels.

Comment (viveksinghggits) [reply web/scripts/display_yaml.js]: would this work .?

File: web/scripts/codemirror/codemirror-yaml.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+(function(mod) {
+    if (typeof exports == "object" && typeof module == "object") // CommonJS
+      mod(require("../../lib/codemirror"));
+    else if (typeof define == "function" && define.amd) // AMD
+      define(["../../lib/codemirror"], mod);
+    else // Plain browser env
+      mod(CodeMirror);
+  })(function(CodeMirror) {
+  "use strict";
+
```
Comment (viveksinghggits) [inline web/scripts/codemirror/codemirror-yaml.js]: remove this, please hosted one.

File: web/scripts/codemirror/codemirror.css [hunk 3]
```diff
@@ -0,0 +1,349 @@
+/* BASICS */
+
+.CodeMirror {
+    /* Set height, width, borders, and global font properties here */
+    font-family: monospace;
+    height: 300px;
+    color: black;
```
Comment (viveksinghggits) [inline web/scripts/codemirror/codemirror.css]: remove this please the hosted one. if we are using code mirror.

File: web/scripts/display_policies.js [hunk 4]
```diff
@@ -167,7 +167,48 @@ function drawArrow(e){
     connect.setAttribute("to", "#"+targetid)
     document.body.appendChild(connect)
 
-    generateNWPolicyYaml(sourceid, targetid)
+    // console.log(document.getElementById(targetid).title)
```
Comment (viveksinghggits) [inline web/scripts/display_policies.js]: have this in entire separate function

File: web/scripts/codemirror/codemirror-yaml.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+      },
+      lineComment: "#",
+      fold: "indent"
+    };
+  });
+  
+  CodeMirror.defineMIME("text/x-yaml", "yaml");
+  CodeMirror.defineMIME("text/yaml", "yaml");
+  
+  });
```
Comment (PrasadG193) [inline web/scripts/codemirror/codemirror-yaml.js]:
```suggestion
  });
  
```

File: web/scripts/display_yaml.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+// // Input parametes for creating the policy yaml
+
+// let source_label = {app: "test", run: "jog"};
+// let target_label = {app: "final"};
+// var port = "8080";
+// var policy_types = "Ingress"
+// var protocol = "TCP"
+// var policy_name = "test-network-policy"
+
+// var doc = jsyaml.load('apiVersion: networking.k8s.io\/v1\r\nkind: NetworkPolicy\r\nmetadata:\r\n  name: test-network-policy\r\n  namespace: default\r\nspec:\r\n  podSelector:\r\n    matchLabels:\r\n      role: db\r\n  policyTypes:\r\n    Ingress\r\n  ingress:\r\n  - from:\r\n    - namespaceSelector:\r\n        matchLabels:\r\n          project: myproject\r\n    - podSelector:\r\n        matchLabels:\r\n          role: frontend\r\n    ports:\r\n    - protocol: TCP\r\n      port: 6379');
```
Comment (viveksinghggits) [inline web/scripts/display_yaml.js:10]: what the fuck is this

**Review by viveksinghggits — CHANGES_REQUESTED** [review]

**Comment by viveksinghggits** [comment]
@rutu-k remove the unnecessary codemirror files or use cdn of them.

File: web/scripts/display_policies.js [hunk 7]
```diff
@@ -167,7 +167,48 @@ function drawArrow(e){
...
+
+    var doc = jsyaml.load('apiVersion: networking.k8s.io\/v1\r\nkind: NetworkPolicy\r\nmetadata:\r\n  name: test-network-policy\r\n  namespace: default\r\nspec:\r\n  podSelector:\r\n    matchLabels:\r\n      role: db\r\n  policyTypes:\r\n  - Ingress\r\n  ingress:\r\n  - from:\r\n    - namespaceSelector:\r\n        matchLabels:\r\n          project: myproject\r\n    - podSelector:\r\n        matchLabels:\r\n          role: frontend\r\n    ports:\r\n    - protocol: TCP\r\n      port: 6379');
+    doc.spec.podSelector.matchLabels = sourceObj;
+    // console.log(doc)
+    doc.spec.ingress[0].from[1].podSelector.matchLabels = targetObj;
+    const yaaml = jsyaml.safeDump(doc);
+    document.getElementById("codegen").value=yaaml
+
+
+    // generateNWPolicyYaml(sourceid, targetid)
```
Comment (PrasadG193) [inline web/scripts/display_policies.js]:
```suggestion
```

File: web/scripts/display_yaml.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+// doc.spec.podSelector.matchLabels = source_label;
+// const yaaml = jsyaml.safeDump(doc);
+// document.getElementById("specs").value=yaaml
+// // console.log(yaaml)
```
Comment (PrasadG193) [inline web/scripts/display_yaml.js]: Remove if not required

File: web/index.html [hunk 9]
```diff
@@ -29,12 +31,24 @@
...
     </div>
 </body>
 <script type="text/javascript" src="scripts/display_policies.js"></script>
 <script type="text/javascript" src="scripts/dnd.js"></script>
+<script type="text/javascript" src="scripts/display_yaml.js"></script>
+<script src="scripts/codemirror/codemirror.js"></script>
+<script type="text/javascript" src="scripts/codemirror/codemirror-yaml.js"></script>
+<!-- <script>
+    document.getElementById("specs").style.resize = "vertical";
+</script> -->
```
Comment (PrasadG193) [inline web/index.html]: Remove if not required

File: web/scripts/display_policies.js [hunk 10]
```diff
@@ -167,7 +167,48 @@ function drawArrow(e){
...
+    let sourceObj = {};
+    for (i=0; i<source_label_split.length; i++) {
+        val = source_label_split[i].split(":")
+        sourceObj[val[0]] = val[1]
+    };
+
+    console.log(targetObj)
+
+
+
```
Comment (PrasadG193) [inline web/scripts/display_policies.js]: Remove unnecessary new lines

File: web/scripts/display_policies.js [hunk 11]
```diff
@@ -167,7 +167,48 @@ function drawArrow(e){
...
+    var port = "8080";
+    var policy_types = "Ingress"
+    var protocol = "TCP"
+    var policy_name = "test-network-policy"
+
+    // console.log(obj)
+
```
Comment (PrasadG193) [inline web/scripts/display_policies.js]: Please remove commented code and white spaces

File: web/scripts/display_yaml.js (hunk 6 above)
Comment (viveksinghggits) [inline web/scripts/display_yaml.js]: do we need this yaml @rutu-k

**Review by viveksinghggits — CHANGES_REQUESTED** [review]

File: web/scripts/display_yaml.js (hunk 6 above)
Comment (rutu-k) [reply web/scripts/display_yaml.js]: removed...

File: web/scripts/display_policies.js [hunk 12]
```diff
@@ -179,8 +179,41 @@ function drawArrow(e){
...
+    source_label_split = source_label.split(",");
+    let sourceObj = {};
+    for (i=0; i<source_label_split.length; i++) {
+        val = source_label_split[i].split(":")
+        sourceObj[val[0]] = val[1]
+    };
 
-    generateNWPolicyYaml(sourceid, targetid)
+    // variables
+    var port = "8080";
```
Comment (viveksinghggits) [inline web/scripts/display_policies.js:207]: we will have to take care of these ports that we are hardcoding.

**Review by viveksinghggits — APPROVED** [review]

File: web/scripts/display_policies.js (hunk 12 above)
Comment (rutu-k) [reply web/scripts/display_policies.js:207]: so that will be user input or we will be getting from workloads

**Pull request merged**